pkg/apis/environment/common.go

```diff
@@ -270,3 +270,26 @@ func validateConnectors(ctx context.Context, mc model.ClientSet, ids []object.ID
 
 	return nil
 }
+
+func setResourceSubjectID(ctx context.Context, mc model.ClientSet, resources ...*model.Resource) error {
```

Comment:
The function name looks identical to pkgresource.SetSubjectID. But the behavior is different. Should we call this updateResourceSubjectID to differentiate?

Reply:
Updated, move it to pkg/resource.